src/zenml/utils/tag_utils.py

```diff
@@ -674,7 +674,7 @@ def remove_tags(
             infer_artifact,
         ]
     ):
-        snapshot_model = client.get_snapshot(id_or_prefix=snapshot)
+        snapshot_model = client.get_snapshot(name_id_or_prefix=snapshot)
```

Comment:
```suggestion
        snapshot_model = client.get_snapshot(name_id_or_prefix=snapshot, allow_prefix_match=False)
```

The current expectation here is that of a full match or nothing.